Userland/Libraries/LibCore/Socket.cpp

```diff
@@ -133,13 +134,43 @@ bool Socket::common_connect(const struct sockaddr* addr, socklen_t addrlen)
                 on_connected();
         }
     };
+
+    auto check_connected_done = [&](const struct sockaddr* addr, socklen_t addrlen) {
+        fd_set wfds {};
+        FD_ZERO(&wfds);
+        FD_SET(fd(), &wfds);
+        if (select(fd() + 1, nullptr, &wfds, nullptr, nullptr) < 0) {
```

Comment:
I'm not sure if replacing a `Notifier::construct(fd(), Notifier::Event::Write, this)` by a `select()` is a good idea here.

My understanding is that the event loop runs a `select` call every time it waits for the next event, and that `Notifier` adds a new file to select on. However, here, using `select` synchronously like this will block the event loop until the TCP select, and only that select, is resolved. In addition, there's no timeout at all to wait for that resolution, meaning that it might block the socket forever if there isn't anything happening on said socket.

Again, though, I'm not totally sure if it would actually be an issue here or if it somehow doesn't apply due to some interaction thing somewhere. I think @alimpfard knows a lot more about this, and would be able to confirm or deny this :^)

Reply:
Yep, this `select()` will block the loop, potentially indefinitely.
Since there already is a write notifier, you could move this connectivity check in the callback for that, the only functional change there would be that you'll end up returning true from connect() for in-progress connections, but no one relies on that as a "is it connected" signal, that's what `on_connected` is for.